server/src/main/scala/io/delta/standalone/internal/DeltaSharedTable.scala

```diff
@@ -645,7 +648,8 @@ class DeltaSharedTable(
       }
     // Return an `endStreamAction` object only when `maxFiles` is specified for
     // backwards compatibility.
-    if (maxFilesOpt.isDefined) {
+    if (maxFilesOpt.isDefined || includeEndStreamAction) {
+      // TODO(Ask why nextPageTokenStr is always null
```

Comment:
Because if we reach here, it means the it's the last page. If it's not the last page, we will return early beforehand.